src/main/java/com/excelsiorjet/api/ExcelsiorJet.java

```diff
@@ -0,0 +1,124 @@
+
+import java.io.File;
+
+/**
+ * This class represents facade for Excelsior JET toolchain.
+ *
+ * @author Aleksey Zhidkov
+ */
+public class ExcelsiorJet {
+
+    private final JetHome jetHome;
+    private final Log logger;
+
+    public ExcelsiorJet(JetHome jetHome, Log logger) {
+        this.jetHome = jetHome;
+        this.logger = logger;
+    }
+
+    public ExcelsiorJet(String jetHome) throws JetHomeException {
+        this(Utils.isEmpty(jetHome) ? new JetHome() : new JetHome(jetHome), Log.logger);
+    }
+
+    /**
+     * Invokes jc command line tool in given working directory with {@code logger} specified in constructor and pass
+     * {@code args} to it.
+     *
+     * @param workingDirectory working directory for jc. May be null, in this case current process's working directory is used.
+     * @param args command line arguments that will be passed to jc.
+     */
+    public int compile(File workingDirectory, String... args) throws CmdLineToolException {
+        return new JetCompiler(jetHome, args)
+                .workingDirectory(workingDirectory)
+                .withLog(logger)
+                .execute();
+    }
+
+    /**
+     * Invokes xpack command line tool in given working directory with {@code logger} specified in constructor and pass
+     * {@code args} to it.
+     *
+     * @param workingDirectory working directory for xpack. May be null, in this case current process's working directory is used.
+     * @param args command line arguments that will be passed to xpack.
+     */
+    public int pack(File workingDirectory, String... args) throws CmdLineToolException {
+        return new JetPackager(jetHome, args)
+                .workingDirectory(workingDirectory)
+                .withLog(logger)
+                .execute();
+    }
+
+    /**
+     * Invokes xjava command line tool in given working directory with {@code logger} specified in constructor and pass
+     * {@code args} to it.
+     *
+     * @param workingDirectory working directory for xjava. May be null, in this case current process's working directory is used.
+     * @param args command line arguments that will be passed to xjava.
+     */
+    public int run(File workingDirectory, String... args) throws CmdLineToolException {
+        return run(workingDirectory, logger, false, args);
+    }
+
+    /**
+     * Invokes xjava command line tool in given working directory, with specified {@code logger} and pass
+     * {@code args} to it.
+     *
+     * @param workingDirectory working directory for xjava. May be null, in this case current process's working directory is used.
+     * @param logger logger, where xjava's output should be redirected.
+     * @param errToOut specifies is xjava's stderr should be redirected with info level.
+     * @param args command line arguments that will be passed to xjava.
+     */
+    public int run(File workingDirectory, Log logger, boolean errToOut, String... args) throws CmdLineToolException {
```

Comment:
It's general method in general API which do not do any thing specific for testRun, so I do not think, that it's name should be derived from specific usage in currently existing clients.